test/instruments_previous_test.go

```diff
@@ -6,96 +6,133 @@ import (
+				assert.NotNil(t, response)
+				assert.Equal(t, instruments.Card, response.Type)
+				assert.NotEmpty(t, response.Id)
+				assert.NotEmpty(t, response.Fingerprint)
+				assert.NotEmpty(t, response.ExpiryMonth)
+				assert.NotEmpty(t, response.ExpiryYear)
+				assert.NotEmpty(t, response.Scheme)
+				assert.NotEmpty(t, response.Last4)
+				assert.NotEmpty(t, response.Bin)
+				assert.NotEmpty(t, response.CardType)
+				assert.NotEmpty(t, response.CardCategory)
+				assert.NotEmpty(t, response.ProductId)
+				assert.NotEmpty(t, response.ProductType)
+				assert.NotEmpty(t, response.Customer)
+				assert.NotEmpty(t, response.Customer.Id)
+				assert.NotEmpty(t, response.Customer.Name)
+				assert.NotEmpty(t, response.Customer.Email)
+			},
+		},
+	}
+
+	client := PreviousApi().Instruments
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.checker(client.Get(tc.responseId))
+		})
+	}
 }
 
 func TestCreateAndUpdateInstrumentPrevious(t *testing.T) {
-
 	cardTokenResponse := RequestCardTokenPrevious(t)
 	createResponse := createTokenInstrumentPrevious(t, cardTokenResponse)
-
 	updateRequest := abc.UpdateInstrumentRequest{
 		ExpiryMonth: 12,
 		ExpiryYear:  2026,
 		Name:        "New Name",
 	}
 
-	updateResponse, err := PreviousApi().Instruments.Update(createResponse.Id, updateRequest)
-	assert.Nil(t, err)
-	assert.NotNil(t, updateResponse)
-	assert.Equal(t, instruments.Card, updateResponse.Type)
-	assert.NotEmpty(t, updateResponse.Fingerprint)
+	cases := []struct {
+		name          string
+		responseId    string
+		updateRequest abc.UpdateInstrumentRequest
+		checker       func(*abc.UpdateInstrumentResponse, error)
+	}{
+		{
+			name:          "when update instrument request then this request is updated",
+			responseId:    createResponse.Id,
+			updateRequest: updateRequest,
+			checker: func(response *abc.UpdateInstrumentResponse, err error) {
+				assert.Nil(t, err)
+				assert.NotNil(t, response)
+				assert.Equal(t, instruments.Card, response.Type)
+				assert.NotEmpty(t, response.Fingerprint)
+				getResponse, err := PreviousApi().Instruments.Get(createResponse.Id)
+				assert.Nil(t, err)
+				assert.NotNil(t, getResponse)
+				assert.Equal(t, instruments.Card, getResponse.Type)
+				assert.Equal(t, 12, getResponse.ExpiryMonth)
+				assert.Equal(t, 2026, getResponse.ExpiryYear)
+				assert.Equal(t, "New Name", getResponse.Name)
+			},
+		},
+	}
 
-	getResponse, err := PreviousApi().Instruments.Get(createResponse.Id)
-	assert.Nil(t, err)
-	assert.NotNil(t, getResponse)
-	assert.Equal(t, instruments.Card, getResponse.Type)
-	assert.Equal(t, 12, getResponse.ExpiryMonth)
-	assert.Equal(t, 2026, getResponse.ExpiryYear)
-	assert.Equal(t, "New Name", getResponse.Name)
+	client := PreviousApi().Instruments
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.checker(client.Update(tc.responseId, tc.updateRequest))
+		})
+	}
 }
 
 func TestCreateAndDeleteInstrumentPrevious(t *testing.T) {
```

Comment:
Change the name to TestDeleteInstrumentPrevious

Reply:
👍